src/tree/v2/application/DefaultApplicationResourceBranchDataProvider.ts

```diff
@@ -0,0 +1,31 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { ApplicationResource, BranchDataProvider } from '../../../api/v2/v2AzureResourcesApi';
+import { localize } from "../../../utils/localize";
+import { ResourceGroupsItem } from '../ResourceGroupsItem';
+import { DefaultApplicationResourceItem } from './DefaultApplicationResourceItem';
+
+export class DefaultApplicationResourceBranchDataProvider implements BranchDataProvider<ApplicationResource, ResourceGroupsItem> {
+    getChildren(element?: ResourceGroupsItem | undefined): vscode.ProviderResult<ResourceGroupsItem[]> {
+        if (!element) {
```

Comment:
I think this check is overkill. IMO in BDP's interface we can override `getChildren` to indicate that `element` will never be falsy, and avoid needing checks like this in every BDP.

Reply:
Yes, your `getChildren()` override somehow never got into the API definition; I've done that and removed the `undefined` checks.